src/contract_client/client.ts

```diff
@@ -45,6 +47,35 @@ export class ContractClient {
     });
   }
 
+  /**
+   * Generate a ContractClient instance from the ContractClientOptions and the wasm binary
+   */
+  static async fromWasm(options: ContractClientOptions, wasm: BufferSource): Promise<ContractClient> {
```

Comment:
nit: the constructor takes `spec` then `options`, whereas this takes `options` then `wasm`. I'd prefer `options` always came in the same order. Second, probably? To avoid the breaking change with making `constructor` take it as the first arg.